packages/block-directory/src/store/resolvers.js

```diff
@@ -36,21 +36,34 @@ export default {
 
 			yield receiveDownloadableBlocks( blocks, filterValue );
 		} catch ( error ) {
-			if ( error.code === 'rest_user_cannot_view' ) {
+			if ( error.code === 'rest_block_directory_cannot_view' ) {
 				yield setInstallBlocksPermission( false );
 			}
 		}
 	},
 	*hasInstallBlocksPermission() {
 		try {
-			yield apiFetch( {
-				path: `__experimental/block-directory/search?term=`,
+			const response = yield apiFetch( {
+				method: 'OPTIONS',
+				path: `__experimental/block-directory/search`,
+				parse: false,
 			} );
-			yield setInstallBlocksPermission( true );
-		} catch ( error ) {
-			if ( error.code === 'rest_user_cannot_view' ) {
-				yield setInstallBlocksPermission( false );
+
+			let allowHeader;
```

Comment:
I think it makes sense to have this here until we are out of `experimental` status and move to: https://github.com/WordPress/gutenberg/blob/f15b91ca9cbdf171448ebc1f23f55cc5b2748262/packages/core-data/src/resolvers.js#L171